File: DeviceManagementSystem/Models/Device.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace DeviceManagementSystem.Models
{
    public class Device
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "箇所名は必須です")]
        public string Location { get; set; }

        [Required(ErrorMessage = "PC/IDは必須です")]
        public string PcId { get; set; }

        public DeviceStatus Status { get; set; }

        public DateTime? ReleaseDeadline { get; set; }

        public bool IsFaultyReplacement { get; set; }
    }

    public enum DeviceStatus
    {
        準備中,
        出荷待ち,
        受取待ち,
        受取済み
    }
}

File: src/device.py
class Device:
    def __init__(self, location_name, pc_id, replacement_flag, status="Preparing", release_deadline=None):
        self.id = id(self)
        self.location_name = location_name
        self.pc_id = pc_id
        self.replacement_flag = replacement_flag
        self.status = status
        self.release_deadline = release_deadline

    def __str__(self):
        return f"ID: {self.id}, Location: {self.location_name}, PC/ID: {self.pc_id}, Replacement: {self.replacement_flag}, Status: {self.status}, Release Deadline: {self.release_deadline}"
